packages/libs/client/src/contPact.ts

```diff
@@ -0,0 +1,477 @@
+import { IPact, NonceType } from '.';
+
+import debug, { Debugger } from 'debug';
+
+const log: Debugger = debug('pactjs:proxy');
+
+/**
+ * @alpha
+ */
+export interface IContCommandBuilder<
+  TCaps extends Record<string, TArgs>,
+  TArgs extends Array<TCaps[keyof TCaps]> = TCaps[keyof TCaps],
+> {
+  createCommand(): IUnsignedCommand;
+  addData: (
+    data: IContCommand['data'],
+  ) => IContCommandBuilder<TCaps, TArgs> & IContCommand;
+  setMeta: (
+    publicMeta: Partial<IContCommand['publicMeta']> & {
+      sender: IContCommand['publicMeta']['sender'];
+    },
+    networkId?: IContCommand['networkId'],
+  ) => IContCommandBuilder<TCaps, TArgs> & IContCommand;
+  addCap<TCap extends keyof TCaps>(
+    caps: TCap,
+    signer: string,
+    ...args: TCaps[TCap]
+  ): IContCommandBuilder<TCaps, TArgs> & IContCommand;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  local(apiHost: string, options?: any): Promise<any>;
+  send(apiHost: string): Promise<SendResponse>;
+  pollUntil(
+    apiHost: string,
+    options?: {
+      interval?: number;
+      timeout?: number;
+      onPoll?: (
+        transaction: IContCommand &
+          IContCommandBuilder<Record<string, unknown>>,
+        pollRequest: Promise<IPollResponse>,
+      ) => void;
+    },
+  ): Promise<this>;
+  poll(apiHost: string): Promise<IPollResponse>;
+  addSignatures(
+    ...sig: {
+      pubKey: string;
+      sig: string;
+    }[]
+  ): IContCommandBuilder<TCaps, TArgs> & IContCommand;
+  setNonceCreator(
+    nonceCreator: (t: IContCommand, dateInMs: number) => NonceType,
+  ): IContCommandBuilder<TCaps, TArgs> & IContCommand;
+  status: string;
+}
+
+type TransactionStatus =
+  | 'malleable'
+  | 'non-malleable'
+  | 'pending'
+  | 'success'
+  | 'failure'
+  | 'timeout';
+
+/**
+ * Used to build Command objects modularly by adding pact code, environment data, capabilities, and sigs
+ * as necessary.
+ * Once the command is complete and ready for use, you can either call `createCommand` to get the ICommand object,
+ * or you can call `local` or `send` to send the command to /send or /local endpoints of a node with the provided URL.
+ *
+ * @alpha
+ */
+export class ContCommand
+  implements IContCommand, IContCommandBuilder<Record<string, unknown>>
+{
+  public data: Record<string, unknown>;
+  public publicMeta: {
+    chainId: ChainId;
+    sender: string;
+    gasLimit: number;
+    gasPrice: number;
+    ttl: number;
+  };
+  public networkId: ChainwebNetworkId;
+  public signers: {
+    pubKey: string;
+    caps: {
+      name: string;
+      args: ICap['args'];
+    }[];
+  }[];
+  public sigs: (ISignatureJson | undefined)[];
+  public type: 'cont' = 'cont';
+  public cmd: string | undefined;
+  public nonce: NonceType | undefined;
+  public proof: string;
+  public step: number;
+  public pactId: string;
+  public rollback: boolean;
+  public status: TransactionStatus;
+  public requestKey: string | undefined;
+
+  public constructor() {
```

Comment:
I think its better to pass (proof, step, pactId, rollback) to the constructor, then you don't need those related setters.
and the API would be more cleaner because we need to set those properties in one step